tempodb/blocklist/poller.go

```diff
@@ -456,6 +466,57 @@ func (p *Poller) tenantIndexPollError(idx *backend.TenantIndex, err error) error
 	return nil
 }
 
+// deleteTenant will delete all of a tenant's objects if there is not a tenant index present.
+func (p *Poller) deleteTenant(ctx context.Context, tenantID string) error {
+	level.Info(p.logger).Log("msg", "deleting tenant", "tenant", tenantID)
+
+	if p.cfg.EmptyTenantDeletionAge == 0 {
+		return fmt.Errorf("empty tenant deletion age must be greater than 0")
+	}
+
+	var (
+		foundObjects  []string
+		recentObjects int
+	)
+	err := p.reader.Find(ctx, backend.KeyPath{tenantID}, func(opts backend.FindMatch) {
+		level.Info(p.logger).Log("msg", "checking object for deletion", "object", opts.Key, "modified", opts.Modified)
+
+		if time.Since(opts.Modified) > p.cfg.EmptyTenantDeletionAge {
+			foundObjects = append(foundObjects, opts.Key)
+		} else {
+			recentObjects++
```

Comment:
This could exit early as soon as it finds the first match, if we added a return value to FindFunc to signal.  Not required for now, but it's keep it in mind if we run into situations where continuing to iterate results is an issue.

Reply:
That's true.  I suspect that the number of objects are in the 10s at most, and the call is in batches of 1000, so likely won't result in additional calls to the backend.  Signaling a stop sounds reasonable if this does come up later.